pd/src/main.rs

```diff
@@ -153,66 +177,285 @@ async fn main() -> anyhow::Result<()> {
+                            max_bytes: 1048576,
                         },
-                        power: tendermint::vote::Power::from(100u32),
+                        validator: tendermint::consensus::params::ValidatorParams {
+                            pub_key_types: vec![Algorithm::Ed25519],
+                        },
+                        version: Some(tendermint::consensus::params::VersionParams {
+                            app_version: 0,
+                        }),
                     },
-                ],
-            };
+                    // always empty in genesis json
+                    app_hash: vec![],
+                    app_state: app_state,
+                    // List of initial validators. Note this may be overridden entirely by
+                    // the application, and may be left empty to make explicit that the
+                    // application will initialize the validator set with ResponseInitChain.
+                    // - https://docs.tendermint.com/v0.32/tendermint-core/using-tendermint.html
+                    // For penumbra, we can leave this empty since the app_state also contains Validator
+                    // configs.
+                    validators: vec![],
+                };
+                let mut genesis_file_path = node_config_dir.clone();
+                genesis_file_path.push("genesis.json");
+                println!(
+                    "Writing {} genesis file to: {}",
+                    &node_name,
+                    genesis_file_path.display()
+                );
+                let mut genesis_file = File::create(genesis_file_path)?;
+                genesis_file
+                    .write_all(serde_json::to_string_pretty(&validator_genesis)?.as_bytes())?;
+
+                // Write this node's config.toml
+                // Note that this isn't a re-implementation of the `Config` type from
+                // Tendermint (https://github.com/tendermint/tendermint/blob/6291d22f46f4c4f9121375af700dbdafa51577e7/config/config.go#L92)
+                // so if they change their defaults or the available fields, that won't be reflected in our template.
+                let tm_config = generate_tm_config(&node_name);
+                let mut config_file_path = node_config_dir.clone();
+                config_file_path.push("config.toml");
+                println!(
+                    "Writing {} config file to: {}",
+                    &node_name,
+                    config_file_path.display()
+                );
+                let mut config_file = File::create(config_file_path)?;
+                config_file.write_all(tm_config.as_bytes())?;
+
+                // Write this node's node_key.json
+                // the underlying type doesn't implement Copy or Clone (for the best)
+                let priv_key: tendermint::PrivateKey;
+                unsafe {
+                    priv_key = mem::transmute_copy(&vk.node_key_sk);
+                }
```

Comment:
I'm OK with this for now, but it would be nice not to use this long-term, maybe we could add a `Clone` impl upstream?